mim/commands/install.py

```diff
@@ -1,580 +1,303 @@
-        echo_success(f'compiling {package} with "MMCV_WITH_OPS=1"')
-        os.environ['MMCV_WITH_OPS'] = '1'
-
-    call_command(install_cmd)
-
-    if is_editable:
-        link_file_to_package()
-    else:
-        copy_file_to_package()
 
+def get_mmdeps_from_mmpkg(mmpkg: str, index_url: Optional[str] = None) -> str:
+    """Get 'mim' extra requirements for a given OpenMMLab package from
+    `mminstall.txt`.
 
-def install_from_github(package: str,
-                        version: str = '',
-                        find_url: str = '',
-                        timeout: int = 15,
-                        is_yes: bool = False,
-                        is_user_dir: bool = False,
-                        is_install_master: bool = False) -> None:
-    """Install package from github.
+    If there is a cached `mminstall.txt`, use the cache, otherwise download the
+    source distribution package from pypi and extract `mminstall.txt` content.
 
     Args:
-        package (str): The name of installed package, such as mmcls.
-        version (str): Version of package. Default: ''.
-        find_url (str): Url for finding package. If finding is not provided,
-            program will infer the find_url as much as possible. Default: ''.
-        timeout (int): The socket timeout. Default: 15.
-        is_yes (bool): Don’t ask for confirmation of uninstall deletions.
-            Default: False.
-        is_usr_dir (bool): Install to the Python user install directory for
-            environment variables and user configuration. Default: False.
-        is_install_master (bool): Whether install master branch. If it is True,
-            process will install master branch. If it is False, process will
-            install the specified version. Default: False.
+        mmpkg (str): The OpenMMLab package name, optionally with a version
+            specifier. e.g. 'mmdet', 'mmdet==2.25.0'.
+        index_url (str, optional): The pypi index url that pass to
+            `get_mminstall_from_pypi`.
+
+    Returns:
+        (str): The text content read from `mminstall.txt`, returns an empty
+            string if anything goes wrong.
     """
-    click.echo(f'installing {package} from {find_url}.')
-
-    _, repo = parse_url(find_url)
-    clone_cmd = ['git', 'clone', find_url]
-    if not is_install_master:
-        clone_cmd.extend(['-b', f'v{version}'])
-
-    with tempfile.TemporaryDirectory() as temp_root:
-        repo_root = osp.join(temp_root, repo)
-        clone_cmd.append(repo_root)
-        call_command(clone_cmd)
-
-        install_from_repo(
-            repo_root,
-            package=package,
-            timeout=timeout,
-            is_yes=is_yes,
-            is_user_dir=is_user_dir)
+    cache_mminstall_dir = os.path.join(DEFAULT_CACHE_DIR, 'mminstall')
+    if not os.path.exists(cache_mminstall_dir):
+        os.mkdir(cache_mminstall_dir)
+    cache_mminstall_fpath = os.path.join(cache_mminstall_dir, f'{mmpkg}.txt')
```

Comment:
Different versions of mmpkg may have different dependencies and versions.

Reply:
good points, done!